EditorPluginExample/Resources/EditorData/Example.plugin.ts

```diff
@@ -0,0 +1,91 @@
+/// <reference path="../../typings/Atomic/AtomicWork.d.ts" />
+/// <reference path="../../typings/Atomic/EditorWork.d.ts" />
+
+const ExamplePluginUILabel = "Example Plugin";
+const ExamplePluginTBPath = "EditorData/Example.tb.txt";
+
+class ExamplePluginService implements Editor.HostExtensions.HostEditorService, Editor.HostExtensions.ProjectService {
+
+    name: string = "ExampleService";
+    description: string = "This service demonstrates plugin functionality functionality.";
+
+    private serviceLocator: Editor.HostExtensions.HostServiceLocator = null;
+    private extensionWindow: Editor.Modal.ExtensionWindow = null;
+
+    private helloLabel: Atomic.UITextField;
+    private nameField: Atomic.UIEditField;
+
+    initialize(serviceLoader: Editor.Extensions.ServiceLoader) {
```

Comment:
Interfaces in TS are not like interfaces in C# or Java.  Interfaces describe the *shape* of an object.  As long as the object being used contains the methods/properties that the callee expects it to have, it will pass.  Even if the caller doesn't explicitly implement the interface.

That being said, I can see where it can be confusing and I think I'm going to go back into the d.ts and see if I can make the interface more appropriate.  The way this is implemented should work though.